common/lib/xmodule/xmodule/js/src/problem/edit.js

```diff
@@ -0,0 +1,763 @@
+                }
+
+                xml = splits.join('');
+
+                // rid white space
+                xml = xml.replace(/\n\n\n/g, '\n');
+
+                // if we've come across demand hints, wrap in <demandhint> at the end
+                if (demandhints) {
+                    demandHintTags.push(demandhints);
+                }
+
+                // make selector to search responsetypes in xml
+                responseTypesSelector = responseTypes.join(', ');
+
+                // make temporary xml
+                // safe-lint: disable=javascript-concat-html
+                $xml = $($.parseXML('<prob>' + xml + '</prob>'));
+                responseType = $xml.find(responseTypesSelector);
+
+                // convert if there is only one responsetype
+                if (responseType.length === 1) {
+                    inputtype = responseType[0].firstElementChild;
+                    // used to decide whether an element should be placed before or after an inputtype
+                    beforeInputtype = true;
+
+                    _.each($xml.find('prob').children(), function(child) {
+                        // we don't want to add the responsetype again into new xml
+                        if (responseType[0].nodeName === child.nodeName) {
+                            beforeInputtype = false;
+                            return;
+                        }
+
+                        if (beforeInputtype) {
+                            // safe-lint: disable=javascript-jquery-insert-into-target
+                            responseType[0].insertBefore(child, inputtype);
+                        } else {
+                            responseType[0].appendChild(child);
+                        }
+                    });
+                    serializer = new XMLSerializer();
+
+                    xml = serializer.serializeToString(responseType[0]);
+
+                    // remove xmlns attribute added by the serializer
+                    xml = xml.replace(/\sxmlns=['"].*?['"]/gi, '');
+
+                    // XMLSerializer messes the indentation of XML so add newline
+                    // at the end of each ending tag to make the xml looks better
+                    xml = xml.replace(/(\<\/.*?\>)(\<.*?\>)/gi, '$1\n$2');
+                }
+
+                // remove class attribute added on <p> tag for question title
+                xml = xml.replace(/\sclass=\'qtitle\'/gi, '');
+                return xml;
+            };
+            responseTypesXML = [];
+            responseTypesMarkdown = markdown.split(/\n\s*---\s*\n/g);
+            _.each(responseTypesMarkdown, function(responseTypeMarkdown) {
+                if (responseTypeMarkdown.trim().length > 0) {
+                    responseTypesXML.push(toXml(responseTypeMarkdown));
+                }
+            });
+            finalDemandHints = '';
+            if (demandHintTags.length) {
+                // safe-lint: disable=javascript-concat-html
```

Comment:
This safe lint suppression (and the one a few lines below) was already present in CoffeeScript. I don't want to get in to safe commit changes as part of this conversion.